src/lib/storage/dictionary_segment.hpp

```diff
@@ -25,55 +25,122 @@ class DictionarySegment : public BaseSegment {
   /**
    * Creates a Dictionary segment from a given value segment.
    */
-  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& base_segment);
+  explicit DictionarySegment(const std::shared_ptr<BaseSegment>& baseSegment) {
+    auto valueSegment = std::static_pointer_cast<ValueSegment<T>>(baseSegment);
+
+    _build_compressed_dictionary(valueSegment->values());
+  }
 
   // SEMINAR INFORMATION: Since most of these methods depend on the template parameter, you will have to implement
   // the DictionarySegment in this file. Replace the method signatures with actual implementations.
 
   // return the value at a certain position. If you want to write efficient operators, back off!
-  AllTypeVariant operator[](const ChunkOffset chunk_offset) const override;
+  AllTypeVariant operator[](const ChunkOffset chunk_offset) const {
+    return static_cast<AllTypeVariant>(_dictionary->at(_attribute_vector->at(chunk_offset)));
+  };
 
   // return the value at a certain position.
-  T get(const size_t chunk_offset) const;
+  T get(const size_t chunk_offset) const {
+      return _dictionary->at(_attribute_vector->at(chunk_offset));
```

Comment:
I'm not sure if this would be better stil but we could also call `value_by_id()`
```suggestion
      return value_by_id(_attribute_vector->at(chunk_offset));
```

Maybe thats a bit more readable...

Reply:
Same as comment below..